exercises/variable-length-quantity/variable_length_quantity_test.go

```diff
@@ -0,0 +1,36 @@
+package variablelengthquantity
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestEncodeDecodeVarint(t *testing.T) {
+	testCases := []struct {
+		input  []byte
+		output uint32
+	}{
+		0: {[]byte{0x7F}, 127},
+		1: {[]byte{0x81, 0x00}, 128},
+		2: {[]byte{0xC0, 0x00}, 8192},
+		3: {[]byte{0xFF, 0x7F}, 16383},
+		4: {[]byte{0x81, 0x80, 0x00}, 16384},
+		5: {[]byte{0xFF, 0xFF, 0x7F}, 2097151},
+		6: {[]byte{0x81, 0x80, 0x80, 0x00}, 2097152},
+		7: {[]byte{0xC0, 0x80, 0x80, 0x00}, 134217728},
+		8: {[]byte{0xFF, 0xFF, 0xFF, 0x7F}, 268435455},
+
+		9:  {[]byte{0x82, 0x00}, 256},
+		10: {[]byte{0x81, 0x10}, 144},
+	}
+
+	for i, tc := range testCases {
+		t.Logf("test case %d - %#v\n", i, tc.input)
+		if o, _ := DecodeVarint(tc.input); o != tc.output {
```

Comment:
Hmm, given that the second return value is being discarded, it could be anything.

Given that in the example implementation it's the number of bytes consumed, now I want to ponder a few things.

These tests expect a `DecodeVarint` that decodes a *single* number it finds in a buffer.

Should we specify (and therefore test) `DecodeVarint`'s behavior:

* If there are multiple numbers in the buffer? `[]byte{0x00, 0x00}`
* If there is not a complete number in the buffer? `[]byte{0x80, 0x80}`
* If the buffer is empty?

Also, as an extension of this funciton that decodes a single varint, should we ask for an implementation of a function that decodes as many numbers as possible from a buffer as well? It would expand the scope of the exercise, though!